composer/models/tasks/classification.py

```diff
@@ -81,15 +82,27 @@ def loss(self, outputs: Tensor, batch: Tuple[Any, Tensor], *args, **kwargs) -> T
         _, targets = batch
         return self._loss_fn(outputs, targets, *args, **kwargs)
 
-    def metrics(self, train: bool = False) -> Union[Metric, MetricCollection]:
-        return self.train_metrics if train else self.val_metrics
+    def get_metrics(self, is_train: bool = False) -> Dict[str, Metric]:
+        if is_train:
+            metrics = deepcopy(self.train_metrics)
+        else:
+            metrics = deepcopy(self.val_metrics)
+
+        if isinstance(metrics, Metric):
+            metrics_dict = {metrics._get_name(): metrics}
+        else:
+            metrics_dict = {}
+            for name, metric in metrics.items():
+                assert isinstance(metric, Metric)
+                metrics_dict[name] = metric
+
+        return metrics_dict
```

Comment:
How do we feel about modifying the `ComposerClassifier` signature to only accept a `MetricCollection` and not a `Union[Metric, MetricCollection]`. Taking it one step further, what about forcing subclasses of `ComposerClassifier` to accept a pre-constructed dictionary? This would resolve all the typing conversions in this function.

Reply:
cc: @Landanjs and @coryMosaicML for some input on how to handle this under the new Metrics API